sunpy/map/mapbase.py

```diff
@@ -1437,13 +1445,53 @@ def cmap(self):
         """
         Return the `matplotlib.colors.Colormap` instance this map uses.
         """
-        cmap = self.plot_settings['cmap']
-        if isinstance(cmap, str):
-            cmap = plt.get_cmap(cmap)
-            # Set the colormap to be this specific instance so we are not
-            # returning a copy
-            self.plot_settings['cmap'] = cmap
-        return cmap
+        return self._cmap
+
+    @cmap.setter
+    def cmap(self, n):
+        if isinstance(n, str):
+            self._cmap = plt.get_cmap(n)
+        else:
+            self._cmap = n
+        self.plot_settings['cmap'] = self._cmap
```

Comment:
Remove this line. After thinking about it a bit, there's no reason we need to continually update `plot_settings` since it will not be used internally to set anything.